pkg/util/cache/cache.go

```diff
@@ -27,6 +27,9 @@ const (
 type Cache []*cacheShard
 
 func NewCache(maxSize int) Cache {
+	if maxSize < shardsCount {
```

Comment:
At any rate, silently returning nil here is not good, it will make the error show up as a crash and not as a construction error. Since no one is checking the result of this to see if it's nil.